docs/Data-flow-in-Site-Kit.md

```diff
@@ -1,3 +1,155 @@
+
+### Step 6: Caching
+
+Most `GET` requests that are reading data, especially in the case of report data, will cache the response in `sessionStorage`. The default cache expiry time is one hour, this is configurable on a case-by-case basis. Some requests, like `GET` requests for account lists, will not cache—this is because a user might update their list of accounts on a service like Google Analytics, then want to refresh their list of accounts in the Site Kit UI. If we cached that data, the user would need to wait one hour to refresh data.
+
+Generally speaking, we cache report data that does not vary (either much or at all) every hour.
+
+If the WordPress REST API returns a non-`2xx` HTTP response code (or the request fails entirely), the client will not save any data for this request in a cache and will instead log the error using a Redux (`@wordpress/data`) action. Usually the client will display this error in an appropriate fashion. Again: failed request data is never cached in `sessionStorage`.
+
+If the data was retrieved from the cache, the cache will not be modified. Cached data is stored with a time-to-live and will be cleared/ignored if "stale".
+
+## Creating a fetch action/selector using `createFetchStore`
+
+When creating an action/selector that triggers a `fetch` request, use the `createFetchStore` function to return a new [data store that can be combined with an existing store](./Data-store-architecture.md).
+
+Here's an example of what a `createFetchStore` looks like, using the existing `core/site/data/connection` route as a reference:
+
+```js
+const fetchGetConnectionStore = createFetchStore( {
+ baseName: 'getConnection',
+ // Make the API request; by default the cache is used for `get` calls.
+ controlCallback: () => {
+  return API.get( 'core', 'site', 'connection' );
+ },
+ // Changes to make to the reducer once the control finishes and gets a response.
+ reducerCallback: ( state, connection ) => {
+  return {
+   ...state,
+   connection,
+  };
+ },
+} );
+```
+
+## Defining REST API endpoints on the server
+
+REST API endpoints are defined for each module, using four methods for the `Module` class. These cover:
+
+1. Set up remote requests (`setup_services`)
+2. REST API routes (`get_datapoint_definitions`)
+3. Data request to make to a 3rd-party API, if needed (`create_data_request`)
+4. Response parsing for the optional data request (`parse_data_response`)
+
+### Set up remote requests
+
+Each module can set up the services that will be used to make third-party requests. The `setup_services` method returns an array of service identifier to service object instances. These "service objects" can be used to interact with a third-party API.
```

Comment:
These are specifically `Google_Service` class instances, other types are not allowed. Accordingly, these are only for communicating with Google APIs.

Reply:
I've clarified that a bit in 73d6fd0cb539, but I don't know if we have good examples of non-Google APIs we communicate with so I left out any generic advice.